src/main/webapp/2Dscript.js

```diff
@@ -292,9 +295,84 @@ function generateCheckBoxEntry(runId) {
   input.id = runId;
   input.addEventListener('click', showPoseData);
   checkBoxEntry.appendChild(input);
+  console.log('The checkbox input is ' + input);
+  runs[runId].checkBox = input;
   return checkBoxEntry;
 }
 
+/**
+ * Generate a color selector with proper event listener for changing color of
+ * map elements.
+ * @param {string} runId
+ * @return {Array<HTMLElement, HTMLElement>}
+ */
+function generateColorPickerEntry(runId) {
+  const colorPickerEntry = document.createElement('td');
+  const colorPicker = document.createElement('input');
+  colorPicker.type = 'color';
+  colorPicker.id = 'color_' + runId;
+  // Initialize each value to a random starting color.
+  colorPicker.value = '#' + Math.floor(
+      Math.random() * 16777215).toString(16);
+  // Change the map graph color whenever a different color is selected.
+  colorPicker.addEventListener('input', (event) => changeRunColor(event));
+  colorPickerEntry.appendChild(colorPicker);
+  runs[runId].color = colorPicker;
+  return colorPickerEntry;
+}
+/**
+ * Change the color for a given run when the colocolorPickerrPicker changes
+ * value.
+ * @param {MouseEvent} event
+ */
+function changeRunColor(event) {
+  const colorId = event.target.id;
+  const runId = colorId.split('_')[1];
+  console.log('Parsed run id is ' + runId);
+  const checkboxValue = runs[runId].checkBox.checked;
+  // Only add the run if the checkbox is checked and we have selected a new
+  // color.
+  if (runs[runId].map !== undefined && checkboxValue) {
+    runs[runId].map.setMap(null);
+    // Create plot with new color;
+    runs[runId].map = plotLine(runs[runId].data, runs[runId].color.value);
+    runs[runId].map.setMap(map);
+  }
+}
+/**
+ * Generate a button to recenter the map on this current run.
+ * @param {string} runId
+ * @return {HTMLElement}
+ */
+function generateViewIcon(runId) {
+  const viewIconEntry = document.createElement('td');
+  const viewIcon = document.createElement('div');
+  viewIcon.id = 'view_' + runId;
+  viewIcon.innerHTML = '<i class=\'fa fa-eye\'></i>';
+  viewIcon.addEventListener('click', function() {
+    const viewId = this.id;  // eslint-disable-line
+    const runId = viewId.split('_')[1];
+    console.log('run id is ' + runId);
+    const isChecked = runs[runId].checkBox.checked;
+    console.log('The checkbox value is ' + isChecked);
+    if (isChecked && runs[runId].data !== undefined &&
+      runs[runId].map !== undefined) {
```

Comment:
Maybe you could pull this logic out into a helper (and re-use it, e.g. in `changeRunColor`):
```
function runIsVisible(run) {
  return run.checkBox.checked && run.data !== undefined && run.map !== undefined;
}
```
Even one-liner helper functions can be beneficial. In this case it relieves the reader from having to piece together the meaning of three different comparisons in order to understand the purpose of the conditional.